src/VisualStudio/Setup.ServiceHub/x64/Roslyn.VisualStudio.ServiceHub.Setup.x64.csproj

```diff
@@ -18,10 +18,29 @@
       <!-- Disable NGEN. Core assemblies are crossgened. -->
       <Ngen>false</Ngen>
     </ProjectReference>
-  </ItemGroup>
+  </ItemGroup> 
   <ItemGroup>
     <None Include="source.extension.vsixmanifest">
       <SubType>Designer</SubType>
     </None>
   </ItemGroup>
+  <!--
+    The properties and targets below will add the VSIXSourceItems (files that go in the VSIX) and the VSIX file itself to the item group that tracks when files are up-to-date.
+    This ensures that these files are recognized in VS to force the VS Extension Project to build when the files are changed, such as prior to debugging.
+    Workaround for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1820277 
+  -->
+  <PropertyGroup>
+    <CollectUpToDateCheckInputDesignTimeDependsOn>$(CollectUpToDateCheckInputDesignTimeDependsOn);AddUpToDateCheckVSIXSourceItems</CollectUpToDateCheckInputDesignTimeDependsOn>
+    <CollectUpToDateCheckOutputDesignTimeDependsOn>$(CollectUpToDateCheckOutputDesignTimeDependsOn);AddUpToDateCheckTargetVsixContainer</CollectUpToDateCheckOutputDesignTimeDependsOn>
+  </PropertyGroup>
+  <Target Name="AddUpToDateCheckVSIXSourceItems" DependsOnTargets="GetVsixSourceItems;Publish">
```

Comment:
No, the paste part is unique to us. I added this because we are shipping certain dependencies in our servicehub vsix, and it's how we collect them into the vsix
https://github.com/dotnet/roslyn/blob/f1a1eaa5730af53fe03c57bd967a803eadb48110/src/VisualStudio/Setup.ServiceHub/x64/Roslyn.VisualStudio.ServiceHub.Setup.x64.csproj#L15

Reply:
that said, I'd love to get rid of the publish step, pls let me know if there's a better way of doing this

Reply:
Not sure it's helpful and maybe @miyanni remembers better than I do, but in project system he set some properties related to deploy during vsix build.

Reply:
As far as I can remember, the only thing related to deployment was the step to allow the VSIX to be used locally:
```xml
<!-- Runs the DeployVsixExtensionFiles from the VSSDK to enable the extension on the local machine. -->
<DeployExtension Condition="'$(CIBuild)' != 'true'">true</DeployExtension>
```

I think for their situation, they basically want to do the opposite of what I did here. I removed certain items from the VSIX, and if they wanted to add certain things, they just change the `Remove` to `Include`.
https://github.com/dotnet/project-system/blob/1e2f0b6f1d0eed84b17ef641269c936050449261/setup/ProjectSystemSetup/ProjectSystemSetup.csproj#L58-L64

Reply:
@MiYanni I think we are talking about different issues. Our problem here is we want to include any dependencies our .NET OOP services need in this vsix (this might even include BCL binaries in certain cases), w/o having to maintain a list manually. Currently this is done by collecting the results of `Publish`, thus the dependency on this target.

Reply:
The confusion is my fault. I didn't understand that context, thought it was something else and brought Yanni in.

Reply:
> @MiYanni I think we are talking about different issues. Our problem here is we want to include any dependencies our .NET OOP services need in this vsix (this might even include BCL binaries in certain cases), w/o having to maintain a list manually. Currently this is done by collecting the results of `Publish`, thus the dependency on this target.

Looked into it. `Microsoft.NET.Publish.targets` runs the `GenerateDepsFile` MSBuild task which is how it calculates the dependencies. So, running Publish and simply including those outputs into your VSIX is definitely the simplest way. The complex way would be calling `GenerateDepsFile` yourself and then reading the contents and including those into your VSIX, which is not worth the risk/effort. https://github.com/dotnet/sdk/blob/9d732b9cf69896f944a238670eed7edf41127af4/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L1076C62-L1076C62

Reply:
Let's take this fix for now to unblock local deployment. If it turns out to be a bigger problem, will take a another look then